modules/collectors.coffee

```diff
@@ -3,16 +3,23 @@ _ = require 'underscore'
 
 load = require "../lib/load"
 modules = require("config").modules
+useWhitelistedKeys = require("config").useWhitelistedKeys
+whitelistedkeys = ''
 
 module.exports = ({app, logger, config}, next) ->
   collectorHandler = (collectors) ->
     return (req, res) ->
+      if useWhitelistedKeys
+        if not _.every(_.keys(req.body), (v) -> _.contains(whitelistedkeys, v))
+          res.send(406, '')
+          return
       res.send(204, '')
 
       for coll in collectors
         coll(req.body, {req, res})
 
   logger.log "Loading collectors: #{ modules.collectors.join(', ') }"
+  whitelistedkeys = "#{modules.whitelistedKeys}".split(',')
```

Comment:
The console will log it like that, but if you try `_.contains(modules.whitelistedKeys, 'file.page.fetchStart')` for example, you'll see it works as you expect.